src/linux/platform_uci.c

```diff
@@ -155,14 +150,152 @@ static bool uci_create_sta(struct radio *radio, struct bssInfo bssInfo)
     return uci_create_iface(radio, bssInfo, false);
 }
 
+/*
+ * policy for UCI get
+ */
+enum {
+    UCI_GET_VALUES,
+    __UCI_GET_MAX,
+};
+
+static const struct blobmsg_policy uciget_policy[__UCI_GET_MAX] = {
+    [UCI_GET_VALUES] = { .name = "values", .type = BLOBMSG_TYPE_TABLE },
+};
+
+/* dlist to store uci wifi-device section names */
+struct uciradiolist {
+    dlist_item l;
+    char *section;
+    char *phyname;
+};
+
+/*
+ * called by UCI get ubus call with all wifi-device sections
+ * populates dlist with uci config names of wifi radios
+ */
+static void radiolist_cb(struct ubus_request *req, int type, struct blob_attr *msg)
+{
+    dlist_head *list = (dlist_head *)req->priv;
+    struct blob_attr *tb[__UCI_GET_MAX];
+    struct blob_attr *cur;
+    struct uciradiolist *item;
+    int rem;
+
+    blobmsg_parse(uciget_policy, __UCI_GET_MAX, tb, blob_data(msg), blob_len(msg));
+
+    if (!tb[UCI_GET_VALUES]) {
+        fprintf(stderr, "No radios found\n");
+        return;
+    }
+
+    blobmsg_for_each_attr(cur, tb[UCI_GET_VALUES], rem) {
+        item = memalloc(sizeof(struct uciradiolist));
+        item->section = strdup(blobmsg_name(cur));
```

Comment:
`item->phyname` remains uninitialized here, which leads to a SEGV in case of error because a `free(uciphymatch->phyname)` is attempted in the `reghandlers_out` path.

I fixed this by replacing the memalloc with a zmemalloc.

Reply:
Yes, well spotted. So using `free(0);` is ok?
Or shouldn't we protect if with an extra `if (uciphymatch->phyname)`?
I also now realized that I'm not freeing the item itself but only the strings created via `strdup()`. Couldn't find `dlist_free()` or anything the like, so you do that by iterating and deleting while iterating?

Reply:
`man 3 free`: "If ptr is NULL, no operation is performed." This is an ISO/ANSI C requirement.

About freeing the items: indeed that is missing. Should be done by iterating with dlist_get_first()/dlist_remove(). I'll take care of it.

Reply:
I updated the PR once again, removing the thus unnecessary `if(...)` before the `free(...)` calls and added a TODO comment about freeing the dlist.